sklearn/linear_model/_least_angle.py

```diff
@@ -476,12 +476,21 @@ def _lars_path_solver(
 
     max_features = min(max_iter, n_features)
 
+    return_dtype = np.float64
+    for input_array in (X, y, Xy, Gram):
+        if input_array is not None:
+            return_dtype = input_array.dtype
+            break
```

Comment:
I which case, I think the tests should be extended to test the case where `X.dtype` is f32 but `y.dtype` is f64.

Reply:
I don't think this case could happen because when the data is validated, X and y dtypes are matched

Reply:
The suggestion is better. I have incorporated the suggestion. Thanks

Reply:
Regarding a case where calling `fit`method with x.dtype=np.float32 and y.dtypes=np.float64,  trained `coef_.dtype` is np.float32.  It may not be desirable. The mismatched case would be addressed later PR.

Reply:
I guess it's the expected outcome. In all estimators where both dtypes are supported, the dtype of y is always matched to the dtype of X.

Reply:
I guess it's the expected outcome. In all estimators where both dtypes are supported, the dtype of y is always matched to the dtype of X.